akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -510,6 +518,20 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  //this limits number of messages that can be processed synchronously during one actor receive.
+  private val shortCircuitLimit: Int = _initial.mat.settings.syncProcessingLimit
+  private var shortCircuitBuffer: ByteBuffer[BoundaryEvent] = _
+  private var currentLimit: Int = shortCircuitLimit
```

Comment:
Isomorphic, but you can't reuse it. If you set throughput to 1000 and you would take that setting here as well, then you would execute 1000*1000 potential events before putting back the thread to the thread-pool. I don't think one of the settings easily carries to the other. The previous heuristic was set so that a graph can roughly execute a full push-pull cycle before giving back the actor the execution (so it can get the next external event from the mailbox if throughput allows it).